tests/Behat/Page/Admin/Product/ShowPage.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Sylius\PriceHistoryPlugin\Behat\Page\Admin\Product;
+
+use Behat\Mink\Element\NodeElement;
+use Sylius\Behat\Page\Admin\Product\ShowPage as BaseShowPage;
+
+class ShowPage extends BaseShowPage implements ShowPageInterface
+{
+    public function openChannelPricingLogEntryIndexPageForSimpleProduct(string $channelName): void
+    {
+        $this->getSimpleProductPricingRow($channelName)->clickLink('Show');
+    }
+
+    public function openChannelPricingLogEntryIndexPageForVariant(string $variantName, string $channelName): void
+    {
+        $this->getVariantPricingRow($variantName, $channelName)->clickLink('Show');
+    }
+
+    protected function getDefinedElements(): array
+    {
+        return array_merge(parent::getDefinedElements(), [
+            'pricing_element' => '#pricing',
+        ]);
+    }
+
+    private function getVariantPricingRow(string $variantName, string $channelName): NodeElement
+    {
+        /** @var NodeElement|null $pricingRow */
+        $pricingRow = $this->getDocument()->find(
+            'css',
+            sprintf('tr:contains("%s") + tr', $variantName),
+        );
+
+        $pricingRow = $pricingRow->find('css', sprintf('td:contains("%s")', $channelName));
```

Comment:
```suggestion
        $pricingRow = $this->getDocument()->find(
            'css',
            sprintf('tr:contains("%s") + tr:contains("%s")', $variantName, $channelName),
        );
```
Wouldn't this work?

Reply:
Also if it does, you could extract it to a named element. most likely the same case with the other method